### PR TITLE
Fix epel-release and pip installation

### PR DESCRIPTION
**Pull request by wadhwa1**
# Description

- Removed epel-release installation. 
This package was required initially to install python-pip (which is pip2 on CentOS machines, refer [this](https://linuxize.com/post/how-to-install-pip-on-centos-7/)). Now that, we are no longer supporting Python2.x on control + managed nodes, removing this installation.
Here's the documentation for python support matrix https://docs.confluent.io/ansible/7.3.0/ansible-requirements.html

- Removed conditional logic to choose between pip package name (for Python3, it would always be python3-pip)

Background/ Reference -
@erikgb's PR https://github.com/confluentinc/cp-ansible/pull/1232
we would need this in 7.2.x branch as well

Fixes # (issue)
https://github.com/confluentinc/cp-ansible/issues/1229

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

On demand job - https://jenkins.confluent.io/job/cp-ansible-on-demand/685/
for debian, ubuntu, rhel molecule scenarios

One caveat with existing tests- 
They do not give real picture as we are using geerlingguy images (it has ansible - which means it already has pip , cryptography installed). We have tested it locally with vanilla docker images 

[testing.zip](https://github.com/confluentinc/cp-ansible/files/9961236/testing.zip)

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/common/tasks/redhat.yml [hunk 1]
```diff
@@ -54,21 +54,16 @@
...
     state: present
   become: true
 
 - name: Upgrade pip
   ansible.builtin.pip:
-    name: "{{ pip_version }}"
+    name: pip
     extra_args: --upgrade
 
 - name: Install cryptography
```
Comment (wadhwa1) [inline roles/common/tasks/redhat.yml:69]: loop over pip_packages

File: roles/common/tasks/debian.yml [hunk 2]
```diff
@@ -213,13 +213,10 @@
...
   tags: package
 
-- set_fact:
-    pip_package_name: "{{ 'python-pip' if ansible_python_version is match('2.*') else 'python3-pip' }}"
-
 - name: Install pip
   apt:
     name:
-      - "{{ pip_package_name }}"
+      - python3-pip
```
Comment (wadhwa1) [inline roles/common/tasks/debian.yml:219]: add host_validation task to check python version

**Review by erikgb — APPROVED** [review]
LGTM, good job! I suggest a dedicated PR for eventual validations of the python version.

File: roles/common/tasks/debian.yml (hunk 2 above)
Comment (wadhwa1) [reply roles/common/tasks/debian.yml:219]: done, added

**Comment by erikgb** [comment]
Still LGTM, even though I would have preferred to have the new validation in a separate PR. It's so much easier to manage the code with atomic commits. Reverts etc....

**Comment by wadhwa1** [comment]
> Still LGTM, even though I would have preferred to have the new validation in a separate PR. It's so much easier to manage the code with atomic commits. Reverts etc....

thanks @erikgb for checking the PR :) 
Ah sorry, I didn't see your earlier comment suggesting dedicated PR for validations. I'll check with team and try to move this in separate.

File: roles/common/vars/main.yml [hunk 3]
```diff
@@ -24,3 +24,7 @@ confluent_yum_repositories:
     gpgcheck: "{{confluent_common_repository_redhat_clients_gpgcheck}}"
     gpgkey: "{{confluent_common_repository_redhat_clients_gpgkey}}"
     enabled: "{{confluent_common_repository_redhat_clients_enabled}}"
+
+### Pip packages to be installed on managed nodes
+pip_packages:
```
Comment (nsharma-git) [inline roles/common/vars/main.yml:29]: the variable which we don't expect to be changed ( generally) should go in common/defaults/main

Comment (wadhwa1) [reply roles/common/vars/main.yml:29]: thanks @nsharma-git for checking, 

yeah, I was confused too, all variables in <role>/defaults/main.yml go in VARIABLES.md documentation. Are we okay with that ? we don't want users to update this list of pip_packages, right ? 🤔

Comment (nsharma-git) [reply roles/common/vars/main.yml:29]: Yes, this is correct. Sorry, I overlooked it.

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**